### PR TITLE
Update MMA to handle `members-data-api` response with user details section added

### PR DESCRIPTION
**Pull request by charleycampbell**
## What does this change?
members data api response body has been refactored so user details section added. A new object has been created called MembersDataApiResponse , this is made up of user (firstName, lastName and email) and products (MembersDataApiItem[]).

This change needs to be released carefully as it relies on the server side change to link up which is why the calls to the API /api/me/mma have been refactored to accept the new object, MembersDataApiResponse, and the current working. 
MembersDataApiResponse has been added to all of the tests where there is a call to /api/me/mma. 

## How to test
On CODE and PROD manually check the screens on the manage-frontend repo and make sure that all the screens and functionality which call out to  /api/me/mma are working.

## How can we measure success?
Success will look like the journeys on manage frontend are not impacted by this change and future tickets in the backlog which require this change are unblocked.

## Have we considered potential risks?
As there is linking up required with a server side change there is a risk that it doesn't work. This can be mitigated with making sure that the first change (i.e. the PR) works on CODE and PROD. Once we are satisfied we can look to release the server side change and make sure that there are no errors or impact on calls to /api/me/mma. Next step is to remove the old object and follow the same steps to check it is working as expected.

## Accessibility

-   [ ] [Tested with screen reader](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#screen-reader)
-   [ ] [Navigable with keyboard](https://github.com/guardian/accessibility/blob/main/people-and-technology/02-physical.md#keyboard)
-   [ ] [Colour contrast passed](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#contrast)
-   [ ] [The change doesn't use only colour to convey meaning](https://github.com/guardian/accessibility/blob/main/people-and-technology/03-visual.md#use-of-colour)

### CODE REVIEW

File: client/fixtures/productDetail.ts [hunk 1]
```diff
@@ -687,3 +694,16 @@ export const cancelledContribution: ProductDetail = {
 	},
 	isTestUser: false,
 };
+
+export const membersDataApiResponse = (
```
Comment (KentGrm) [inline client/fixtures/productDetail.ts:698]: This is a transformation function, yes? Might read better as `toMembersDataApiResponse` ?

**Review by KentGrm — COMMENTED** [review]
Just checking it's thoroughly tested with both old and new response types?

File: client/__tests__/components/productResponse.test.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
+/* eslint jest/no-standalone-expect: "off", jest/no-done-callback: "off" */
```
[the remark this answers is not in the document]
Comment (charleycampbell) [reply client/__tests__/components/productResponse.test.tsx]: Good spot! This was causing the Team City build to fail so I have removed it 👍

**Comment by charleycampbell** [comment]
@KentGrm would you prefer if I didn't refactor all of the tests to use the new object to ensure we have test coverage for the current and new response? I could revert some of the test changes and then change them all to use the new response type as part of the clean up

**Comment by KentGrm** [comment]
> @KentGrm would you prefer if I didn't refactor all of the tests to use the new object to ensure we have test coverage for the current and new response? I could revert some of the test changes and then change them all to use the new response type as part of the clean up

I would add in a couple of basic tests to prove it handles them both, for say one component. So have both tests side by side

File: client/components/mma/billing/Billing.tsx [hunk 3]
```diff
@@ -45,14 +47,16 @@ type MMACategoryToProductDetails = {
 };
 
 class BillingDataAsyncLoader extends AsyncLoader<
-	[MembersDataApiItem[], { invoices: InvoiceDataApiItem[] }]
+	[MembersDataApiResponse, { invoices: InvoiceDataApiItem[] }]
```
[the remark this answers is not in the document]
Comment (charleycampbell) [reply client/components/mma/billing/Billing.tsx:50]: good spot I've added it in 👍

**Comment by charleycampbell** [comment]
@KentGrm there is a test in productResponse.test.tsx which checks that we accept the current and new response type. I've changed two tests in AccountOverview.stories.tsx back to accept the current array of products so there are now tests in that file which accepts both ways. Let me know if you had other ideas

**Review by KentGrm — APPROVED** [review]

**Comment by KentGrm** [comment]
> @KentGrm there is a test in productResponse.test.tsx which checks that we accept the current and new response type. I've changed two tests in AccountOverview.stories.tsx back to accept the current array of products so there are now tests in that file which accepts both ways. Let me know if you had other ideas

Looks good. Go!

**Comment by charleycampbell** [comment]
[Update MMA to handle `members-data-api` response with user details section added](https://trello.com/c/cikeI3UT/3713-update-mma-to-handle-members-data-api-response-with-user-details-section-added)

**Pull request merged**